[clamped(to:)] Update proposal per Swift Forums thread feedback (#1156)
apple/swift-evolution: proposals/0177-add-clamped-to-method.md

> # Add clamped(to:) to the stdlib

Reviewer: ```suggestion
# Add `clamped(to:)` methods to `Comparable`
```

Author: @benrimmington Thanks!

> 

Reviewer: ```suggestion
* Status: **Awaiting review**
* Implementation: [apple/swift#32268](https://github.com/apple/swift/pull/32268)
* Previous Revision: [1](https://github.com/apple/swift-evolution/blob/8d9f51f060b8c772f7adabee486112a97569d2b3/proposals/0177-add-clamped-to-method.md)

```

## Introduction

This proposal aims to add functionality to the standard library for clamping a value to a provided range.
The proposed function would allow the user to specify a range to clamp a value to where if the value fell within the range, the value would be returned as is, if the value being clamped exceeded the upper or lower bound then the upper or lower bound would be returned respectively.

> Swift-evolution thread: [Add a `clamp` function to Algorithm.swift](https://lists.swift.org/pipermail/swift-evolution/Week-of-Mon-20170306/thread.html#33674)

Reviewer: ```suggestion
Swift-evolution thread: [Add a `clamp` function to Algorithm.swift](https://forums.swift.org/t/add-a-clamp-function-to-algorithm-swift/5405)
```

There already exists an extension to `CountableRange` in the standard library  implementing `clamped(to:)` that will limit the calling range to that of the provided range, so having the same functionality but just for types that conform to the `Comparable` protocol would be conceptually consistent.

> Having functionality like `clamped(to:)` added to `Comparable` as a protocol extension would benefit users of the Swift language whom wish to guarantee that a value is kept within bounds, perhaps one example of this coming in handy would be to limit the result of some calculation between two acceptable numerical limits, say the bounds of a normalized coordinate system.

Reviewer 2: I think `who` will be suitable instead of `whom`

```suggestion
Having functionality like `clamped(to:)` added to `Comparable` as a protocol extension would benefit users of the Swift language who wish to guarantee that a value is kept within bounds, perhaps one example of this coming in handy would be to limit the result of some calculation between two acceptable numerical limits, say the bounds of a normalized coordinate system.
```
